cors.go

```diff
@@ -79,7 +184,17 @@ func AllowedHeaders(headers []string) CORSOption {
 // AllowedMethods ...
 func AllowedMethods(methods []string) CORSOption {
 	return func(ch *cors) error {
-		ch.allowedMethods = methods
+		for _, v := range methods {
+			normalizedMethod := strings.ToUpper(strings.TrimSpace(v))
+			if normalizedMethod == "" {
+				continue
+			}
+
+			if !ch.isMatch(normalizedMethod, ch.allowedMethods) {
+				ch.allowedHeaders = append(ch.allowedHeaders, normalizedMethod)
```

Comment:
Sounds good to me. As long as we document it (and that AllowedHeaders
*appends*) it should be OK!
On Thu, 24 Dec 2015 at 8:38 AM, Nicholas Johns <notifications@github.com>
wrote:

> In cors.go
> <https://github.com/gorilla/handlers/pull/58#discussion_r48389344>:
>
> > @@ -79,7 +184,17 @@ func AllowedHeaders(headers []string) CORSOption {
> >  // AllowedMethods ...
> >  func AllowedMethods(methods []string) CORSOption {
> >  	return func(ch *cors) error {
> > -		ch.allowedMethods = methods
> > +		for _, v := range methods {
> > +			normalizedMethod := strings.ToUpper(strings.TrimSpace(v))
> > +			if normalizedMethod == "" {
> > +				continue
> > +			}
> > +
> > +			if !ch.isMatch(normalizedMethod, ch.allowedMethods) {
> > +				ch.allowedHeaders = append(ch.allowedHeaders, normalizedMethod)
>
> I couldn't see anything in the spec that says we have to support it only
> that a request qualifies as a simple request if its method matches. So I'm
> committing a change that makes AllowedMethods a replacement instead of an
> append operation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gorilla/handlers/pull/58/files#r48389344>.
>